File: Array/leetcode_3131.cpp

// Problem Link ====>> https://leetcode.com/problems/find-the-integer-added-to-array-i/description/

#include<bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int addedInteger(vector<int> &nums1, vector<int> &nums2)
    {
        sort(nums1.begin(), nums1.end());
        sort(nums2.begin(), nums2.end());
        return nums2[0] - nums1[0];
    }
};

int main(){

    Solution s;
    int n;
    cin >> n;
    vector<int> nums1;
    vector<int> nums2;
    cout << "Enter elements for the first array:\n";
    for (int i = 0; i < n; ++i)
    {
        cin >> nums1[i];
    }

    cout << "Enter elements for the second array:\n";
    for (int j = 0; j < n; ++j)
    {
        cin >> nums2[j];
    }

    int x = s.addedInteger(nums1, nums2);

    cout << x << endl;
    return 0;
}

File: Array/leetcode_3131.py

# Problem Link ====>> https://leetcode.com/problems/find-the-integer-added-to-array-i/description/

class Solution:
    def addedInteger(self, nums1: list[int], nums2: list[int]) -> int:
        nums1.sort()
        nums2.sort()
        return nums2[0] - nums1[0]


if __name__ == "__main__":
    solution = Solution()
    
    n = int(input("Enter the size of both arrays: "))

    nums1 = []
    nums2 = []

    print("Enter elements for the first array:")
    for i in range(n):
        num = int(input())
        nums1.append(num)

    print("Enter elements for the second array:")
    for i in range(n):
        num = int(input())
        nums2.append(num)

    difference = solution.addedInteger(nums1, nums2)

    print("Difference between smallest elements:", difference)
